cypress/support/po/components/Dashboard.js

```diff
@@ -0,0 +1,111 @@
+import {BasePage} from "../pages/BasePage";
+
+export class Dashboard extends BasePage{
+    constructor() {
+        super()
+        this.editedDashboardName = ''; // Initialize a variable to store the edited dashboard name
+    }
+    get 'successfully signin msg'() {
+        return cy.contains('Signed in successfully');
+    }
+
+    get 'add new dashboard button'() {
+        return cy.get('.addDashboardButton__add-dashboard-btn--acseh button');
+    }
+
+    get 'new dashboard name from popup'() {
+        return cy.get('[placeholder="Enter dashboard name"]');
+    }
+
+    get 'new dashboard name from description'() {
+        return cy.get('textarea[placeholder="Enter dashboard description"]');
+    }
+
+    get 'add dashboard button from popup'() {
+        return cy.get('button[class^="bigButton__big-button"]').contains('Add');
+    }
+
+    get 'dashboard added successfully'() {
+        return cy.contains('Dashboard has been added');
+    }
+
+    get 'dashboard deleted successfully'() {
+        return cy.contains('Dashboard has been deleted');
+    }
+
+    get 'edit dashboard button'() {
+        return cy.contains('Edit');
+    }
+
+    get 'update button from edit popup'() {
+        return cy.get('button[class^="bigButton__big-button"]').contains('Update');
+    }
+
+    get 'delete dashboard button'() {
+        return cy.get('button[class^="ghostButton__ghost-button"]').contains('Delete');
+    }
+
+    get 'full screen dashboard button'() {
+        return cy.get('button[class^="ghostButton__ghost-button"]').contains('Full screen');
+    }
+
+    get 'add new widget'() {
+        return cy.get('button[class^="ghostButton__ghost-button"]').contains('Add new widget');
+    }
+
+    get 'dashboard name from menu'() {
+        return cy.get('ul[class^="pageBreadcrumbs__page-breadcrumbs"] span');
+    }
+
+    get 'confirm delete button'() {
+        return cy.get('button[class^="bigButton__big-button"]').contains('Delete');
+    }
+
+    addNewDashboard() {
+        this['add new dashboard button'].click();
+    }
+
+    enterUniqueDashboardName(baseName) {
+        const timestamp = new Date().getTime(); // Get current timestamp
```

Comment:
Unique name generation must be extracted to the utility file, because you reuse this functionality in 2 places already.